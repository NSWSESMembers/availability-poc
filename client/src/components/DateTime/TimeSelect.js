import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import { isSelectorDisabled, selectColor } from '../../selectors/schedules';

import { ButtonBox } from '../Button';

const TimeSelect = ({ selectionSegments, onPress }) => {
  const start = moment()
    .startOf('day')
    .unix();

  return (
    <View style={{ flexDirection: 'row', justifyContent: 'space-around', flexWrap: 'wrap' }}>
      {selectionSegments.map(segment => (
        <View key={`${segment.startTime}-${segment.endTime}`} style={{ marginBottom: 5 }}>
          <ButtonBox
            text={segment.label}
            subtext={`${moment.unix(start + segment.startTime).format('HH:mm a')} - ${moment
              .unix(start + segment.endTime)
              .format('HH:mm a')}`}
            onPress={() => onPress(segment)}
            selected={segment.status !== ''}
            selectedColor={selectColor(segment.status)}
            disabled={isSelectorDisabled(
              selectionSegments,
              segment.status,
              segment.startTime,
              segment.endTime,
            )}
          />
        </View>
      ))}
    </View>
  );
};

TimeSelect.propTypes = {
  onPress: PropTypes.func.isRequired,
  selectionSegments: PropTypes.arrayOf(
    PropTypes.shape({
      startTime: PropTypes.number.isRequired,
      endTime: PropTypes.number.isRequired,
      label: PropTypes.string.isRequired,
    }),
  ),
};

export default TimeSelect;
