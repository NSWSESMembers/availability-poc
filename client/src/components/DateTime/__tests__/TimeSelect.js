import React from 'react';
import renderer from 'react-test-renderer';

import TimeSelect from '../TimeSelect';

const noop = () => {};

const testData = {
  startTime: 1,
  endTime: 2,
  label: 'test',
};

test('renders correctly', () => {
  const tree = renderer.create(
    <TimeSelect
      selectionSegments={[testData]}
      onPress={noop}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
