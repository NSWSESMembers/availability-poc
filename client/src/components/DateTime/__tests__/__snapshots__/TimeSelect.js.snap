// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "margin": "1%",
        "width": "48%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        test
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        00:00 - 00:00
      </Text>
    </View>
  </View>
</View>
`;
